### ISSUE TITLE
New namespace for commands?

### ISSUE THREAD

**Issue by ionut-tanasa** [andersao/l5-repository#62]
I was thinking last night about the namespace for the generators.

Since you did an awesome job with this package and I want to contribute as much as I can. 

Yes we have the commands to generate the repositories, the model and the presenter.

Another cool thing would be to generate the provider too that binds the contract to the implementation in the container. So the user has to include it among the apps providers. Maybe we could append directly the created service provider to the providers array in app.php?

Another thing could be generating the controller also for the given resource. With the repository injected. And all the methods set up already. 

What about routes? We could append the newly created resource to the routes file too?

As a developer I tend to try and write the least boilerplate I can during development and concentrate on the business logic and all that stuff. Making a single command that generates everything and sets it up for me would be great.

So to recap, the flow would be like this:

 * create model
 * create repositories
 * create service provider for the repositories
 * append the newly created service provider to the providers array in app.php
 * create the presenter
 * create the transformer
 * create the RESTful controller with the injected repository contract and controller methods already filled up.
 * append the resource route to the routes file
 * create a migration class ?
 * create a seeder class ?
 * finally maybe call the optimize command

Namespacing the commands, for example, under 'prettus', we could do prettus:provider, prettus:controller without conflicting with existing Laravel commands.

What do you think about this?

**Comment by andersao** [comment]
I've been very busy in recent months with my work projects, which eventually pulling away of my personal projects, so it could not provide the necessary support to you, I am more relaxed now and I'm checking all the issues in order to work on fixes and new implementations, apologize to everyone.
